Introduce troubleshooting guide

borine opened the pull request:
Perhaps troubleshooting tips would be better placed in a separate file? I thought I'd try it just to see what other people think. I've made each item a heading, so that each can be directly linked, e.g. when commenting on issues.

arkq commented on TROUBLESHOOTING.md:
```diff
@@ -0,0 +1,82 @@
+
+## Using BlueALSA alongside PulseAudio or PipeWire
+
+Due to BlueZ limitations, only one program can register as provider of
```
I think, that maybe it should be worded differently. When I was writing this, my knowledge about BlueZ was rather limited :D

Firstly, it's not BlueZ limitation, but a design. Secondly, only one program **should** register if one would like to have deterministic behavior. And thirdly, for A2DP it's not true that it's not possible to run BlueALSA with PulseAudio/PipeWire (with their BT modules enabled). The case is that in such setup first Bluetooth device will be connected with an application which has registered itself before other apps.

So, maybe something like this would be better:

> It is not advised to run BlueALSA if either PulseAudio or PipeWire are also running with their own
Bluetooth modules enabled. If one would like to have a deterministic setup, it is first necessary to disable Bluetooth in those applications.

arkq commented:
And one more thought. Maybe we should add a hint about TROUBLESHOOTING.md at the beginning of the bug report issue template?

borine commented:
> Maybe we should add a hint about TROUBLESHOOTING.md at the beginning of the bug report

I've added a link to the troubleshooting guide to the template, and extended the guide with issue reporting hints. Also added an INSTALL.md section on manually setting up the runtime environment.

borine commented on INSTALL.md:
```diff
@@ -64,29 +64,120 @@ configuration):
+state storage. This directory should be called `bluealsa` and be located under
+the system local state directory, which is normally `/var/lib`. The directory
+owner must be the user account that the `bluealsa` daemon is run under, and
+permissions should be `rwxr-xr-x`. For example, on a standard file hierarchy,
```
I've changed the advice to recommend 0700 permissions for local state directory.

borine commented on TROUBLESHOOTING.md:
```diff
@@ -0,0 +1,194 @@
+[previous issues]: https://github.com/arkq/bluez-alsa/issues
+[wiki]: https://github.com/arkq/bluez-alsa/wiki
+
+If reporting a problem as a new issue, please clearly state:
```
I've shifted some of the text out of TROUBLESHOOTING and into README. Hopefully this now works better.

arkq reviewed (approved):
Now it looks great. Thanks!
